quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs

```diff
@@ -409,6 +379,57 @@ instance forall state. StateModel state => Arbitrary (Actions state) where
               | otherwise = ps : map (p :) (go ps)
          in go acts
 
+newtype MoreActions state = MoreActions {getMoreActions :: Actions state}
+
+instance Show (Actions state) => Show (MoreActions state) where
+  show = show . getMoreActions
+
+instance StateModel state => Arbitrary (MoreActions state) where
+  arbitrary = MoreActions <$> generateActionsWithOptions (defaultGenActionsOptions{genOptLengthMult = 10})
+  shrink (MoreActions as) = MoreActions <$> shrink as
+
+-- NOTE: indexed on state for forwards compatibility, e.g. when we
+-- want to give an explicit initial state
+data GenActionsOptions state = GenActionsOptions {genOptLengthMult :: Int}
```

Comment:
Seems like YAGNI to me: is this type exposed externally to the library? If not, then changing it add a `state` type variable later won't make it a breaking change so it's fine to have a simpler type right now

Reply:
I want to expose this interface to make it convenient to work around the defaults in `MoreActions` so I think we should expose this from the start - hence the comment. I may have neglected to do so however, which is my bad.